### DISCUSSION TITLE
setFocus returning undefined with MUI V7

### DISCUSSION THREAD

**Discussion by GoofyGlenn** [react-hook-form/react-hook-form#5652] [Q&A] [answered]
Hi, I am a junior developer, I am currently trying to use the setFocus function on a MUI TextField.

the problem is that the setFocus returns _undefined_ and I am not sure how it should be properly set. I have tried to use the example shown in the  [https://react-hook-form.com/api/useform/setFocus](url)

code shown below:

`interface formInput { 
title: string}`

` const { handleSubmit, control, formState: { errors }, reset, register , setFocus, getValues } = useForm<formInput >({  });`

` useEffect(() => { 
setFocus("title");
})`

```
<FormControl fullWidth={true}>
                            <Controller
                                render={({ field }) =>
                                <TextField label={"titleField"}{...field} error={errors.title !== undefined} helperText={errors.title?.message}  /> }
                                name="title"
                                control={control}
                                defaultValue=""
                                rules={{ required: "this field is required" }}
                            />
                        </FormControl>

```

**Comment by bluebill1049** [comment]
can you provide a codesandbox?

**Reply by GoofyGlenn** [reply]
https://codesandbox.io/s/jolly-wozniak-eg9vu?file=/src/demo.tsx hopefully this demonstrates it. it's my first time really using codesandbox

**Comment by bluebill1049** [comment, marked as answer]
it's inputRef instead of ref for MUI: https://codesandbox.io/s/goofy-liskov-tr3uu

**Reply by GoofyGlenn** [reply]
thank you very much.  it's not always easy to tell with MUI. and again, thank you
